lightning/src/ln/channelmanager.rs

```diff
@@ -1537,31 +1536,31 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					// hopefully an attacker trying to path-trace payments cannot make this occur
 					// on a small/per-node/per-channel scale.
 					if !chan.is_live() { // channel_disabled
-						break Some(("Forwarding channel is not in a ready state.", 0x1000 | 20, Some(self.get_channel_update(chan).unwrap())));
+						break Some(("Forwarding channel is not in a ready state.", 0x1000 | 20, Some(self.get_channel_update_for_unicast(chan).unwrap())));
 					}
 					if *amt_to_forward < chan.get_counterparty_htlc_minimum_msat() { // amount_below_minimum
```

Comment:
why send a channel update to the counterparty if the *counterparty* policy was violated?

Reply:
Are you referring to the `amt_to_forward` value? It looks like the `htlc_minimum_msat` field in the generated channel update is `chan.get_counterparty_htlc_minimum_msat()`.

Reply:
whoa, this got attached to the wrong line, sorry

Reply:
No, this is being sent back to the HTLC sender. Practically its a no-op because the HTLC-sender shouldn't really store info about a private channel, but the spec says we have to include a channel_update for this error type. In reality, the bug is on the recipient's end, they shouldnt have included the channel hint in the invoice to begin with.

Reply:
ah, got it now.  thank you.